modules/p2/ding_serendipity/plugins/content_types/serendipity_ting_object.inc

```diff
@@ -88,15 +88,10 @@ function ding_serendipity_serendipity_ting_object_content_type_render($subtype,
       'type' => 'setting',
     );
 
-    $tmp = array();
-    foreach ($content as $key => $item) {
-      $item['#weight'] = $key;
-      $tmp[] = $item;
-    }
```

Comment:
The items already had a weight, set in the ding_serendipity_render() function. This weight was already matching the index in the array.